### PR TITLE
Study groups

### LINKED ISSUE #314: Build "Join a Study Group" tab
https://redpen.io/ow3163da6caa22acde

- [ ] Needs intro content
- [ ] Might need help from @auremoser for quicker built out of map
- [ ] Need content from API

### PR DESCRIPTION
**Pull request by alanmoo**
Closes #314, #315

### CODE REVIEW

**Comment by gvn** [comment]
Something weird is happening w the collapse/expand transition in Chrome. The element jumps over to the side and then jumps back...

**Comment by gvn** [comment]
FWIW: I don't think columns are absolutely necessary.

File: app/react/pages/study-groups/study-groups.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  .jumbotron-fluid {
+    @include img-retina('/assets/img/hero-studygroup.jpg','/assets/img/hero-studygroup@2x.jpg', cover, null);
+    background-repeat: no-repeat;
+    background-size: cover;
+  }
+  .rc-collapse {
+    columns: 2;
+    -moz-columns: 1;
+  }
+  .rc-collapse * {
```
Comment (gvn) [inline app/react/pages/study-groups/study-groups.scss]: Nest inside previous selector?

File: app/react/pages/study-groups/join-study-group.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import Collapse from "rc-collapse";
+let Panel = Collapse.Panel;
+
+export default React.createClass({
+  propTypes:{
+    switchTabs: React.PropTypes.func.isRequired
+  },
+  componentWillMount() {
+    Service.studyGroups
+      .get()
```
Comment (gvn) [inline app/react/pages/study-groups/join-study-group.jsx]: does this need format: json?

File: app/react/pages/study-groups/join-study-group.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      <div className="container">
+        <div className="row">
+          <div className="col-xs-12 col-lg-10 col-lg-push-1">
+            <h3>What is a Study Group?</h3>
+            <p className="m-b-3">A Study Group is a community of peers committed to learning and teaching each other. They’re fun, informal meetups allowing participants to share skills, experiences, and ideas around open science, open source, code, and community in research. The goal of the Mozilla Study Group Project is to support this kind of peer-to-peer study by providing a simple set of tools, template lesson plans, and access to an international community of like-minded researchers and avid learners in code.</p>
+            <h3>Find a Study Group</h3>
+            <div className="text-xs-center m-y-3">
+              <img src="/assets/img/map.png" srcSet="/assets/img/map.png, /assets/img/map@2x.png 2x" alt="Map of study groups around the world"/>
+            </div>
+            <style>{`.rc-collapse-item:nth-of-type(${middle}){break-after: column; border-bottom-width: 1px}`}</style>
```
Comment (gvn) [inline app/react/pages/study-groups/join-study-group.jsx]: 😵

**Comment by gvn** [comment]
Looking pretty good. Need to address that animation glitch though. If we have to roll with 1 column layout let's just do that...

File: app/react/pages/study-groups/join-study-group.jsx (hunk 3 above)
Comment (alanmoo) [inline app/react/pages/study-groups/join-study-group.jsx]: I didn't feel like it was worth the time to try and upstream dynamic classNames to the rc-collapse module

File: app/react/pages/study-groups/join-study-group.jsx (hunk 2 above)
Comment (alanmoo) [inline app/react/pages/study-groups/join-study-group.jsx]: No, gets that from default params in backend.js. Only need it if adding extra params.

Comment (gvn) [inline app/react/pages/study-groups/join-study-group.jsx]: Gotcha

**Comment by alanmoo** [comment]
Animation is fixed and I cleaned up/paginated the run-study-group page.

**Comment by alanmoo** [comment]
I guess Travis has started drinking early for a Friday afternoon.

**Comment by gvn** [comment]
A couple more things, then R+

- [ ] Study group lesson headers are set to `href="#"`, is there data for the actual targets?
- [ ] Is Location required for form submit? Currently I can leave it blank.

**Comment by gvn** [comment]
sk8

**Pull request merged**